src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  selectedIDs: number[];
+  handleDeleteTodo: (id: number) => void;
+  handleEditTodo: (id: number, data: Partial<Todo>) => void;
+};
+
+export const TodoItem: React.FC<Props> = React.memo(({
+  todo,
+  selectedIDs,
+  handleDeleteTodo,
+  handleEditTodo,
+}) => {
+  const newTodoField = useRef<HTMLInputElement>(null);
+  const [isEditing, setIsEditing] = useState(false);
+  const [currentTitle, setCurrentTitle] = useState(todo.title);
+
+  const toggleStatus = () => {
+    const newData = {
+      completed: !todo.completed,
+    };
+
+    handleEditTodo(todo.id, newData);
+  };
+
+  const editTodoTitle = (event: React.FormEvent) => {
```

Comment:
https://user-images.githubusercontent.com/110383794/201534703-395364d3-dc09-4c96-b32a-2da7a27687ea.mp4

After editing the title of the todo and clicking 'Escape', canceled title is saved in the input field. And if there are no letters in input it's better to show a user a message that the todo will be deleted.